### PR TITLE
Add Enter/LeaveAlternateScreen commands

### PR DESCRIPTION
**Pull request by zrzka**
* Internal refactoring (no breaking changes)
  * Just to follow the `-cursor`, etc. code style/structure
* New commands introduced `EnterAlternateScreen`, `LeaveAlternateScreen`
* Tested
  * Tested on macOS -> `AnsiAlternateScreen`
  * Didn't test on Windows -> `WinApiAlternateScreen` -> @TimonPost please test on Windows

---

Tested with ...

```rust
use std::io::{stdout, Write};
use std::thread::sleep;
use std::time::Duration;

use crossterm_screen::{
    execute, AlternateScreen, EnterAlternateScreen, LeaveAlternateScreen, Result,
};

macro_rules! pause {
    () => {
        sleep(Duration::from_secs(2));
    };
}

fn test_commands() -> Result<()> {
    execute!(stdout(), EnterAlternateScreen)?;
    println!("test_commands: Alternate");
    pause!();
    execute!(stdout(), LeaveAlternateScreen)?;
    println!("test_commands: Left alternate");
    Ok(())
}

fn test_struct() -> Result<()> {
    let _alternate = AlternateScreen::to_alternate(false)?;
    println!("test_struct: Alternate");
    pause!();
    Ok(())
}

fn test_struct_raw() -> Result<()> {
    let _alternate = AlternateScreen::to_alternate(true)?;
    println!("test_struct_raw: Alternate");
    println!("Type something, shouldn't be echoed");
    pause!();
    Ok(())
}

fn main() -> Result<()> {
    println!("Main");
    pause!();
    test_commands()?;
    println!("Back on main after test_commands");
    pause!();
    test_struct()?;
    println!("Back on main after test_struct");
    pause!();
    test_struct_raw()?;
    println!("Back on main after test_struct_raw");
    Ok(())
}
```

### CODE REVIEW

**Comment by zrzka** [comment]
@TimonPost please, test this properly on Windows. I didn't install all the tools on my Windows 10 inside VMware yet.

@Canop just FYI ^ (first comment)

File: src/lib.rs [hunk 1]
```diff
@@ -64,7 +64,168 @@
...
+
+pub use self::raw::{IntoRawMode, RawScreen};
+
+mod alternate;
+mod raw;
 mod sys;
+
+/// An alternate screen.
+///
+/// With this type you will be able to switch to the alternate screen and then back to
```
Comment (TimonPost) [inline src/lib.rs]:
```suggestion
/// With this type, you will be able to switch to the alternate screen and then back to
```

File: src/lib.rs [hunk 2]
```diff
@@ -64,7 +64,168 @@
...
+        let alternate = alternate::alternate_screen();
+        alternate.enter()?;
+
+        let mut alternate = AlternateScreen {
+            alternate,
+            raw_screen: None,
+        };
+
+        if raw_mode {
+            // If into_raw_mode fails, alternate will be dropped and
```
Comment (TimonPost) [inline src/lib.rs]:
```suggestion
            // If into_raw_mode fails, `alternate` will be dropped and
```

**Review by TimonPost — COMMENTED** [review]
Where is `Screen`, if you removed it, can you add it to the changelog as well?

**Review by zrzka — COMMENTED** [review]
Which `Screen`? Didn't remove it, just comments about non existent `Screen`.

**Comment by TimonPost** [comment]
I tested it on windows, and it works. I Meen the `Screen` type, I can't find it. I think you have renamed it to `AlternateScreen` instead.

**Comment by zrzka** [comment]
Thanks for the Windows tests!

Check the diff. I didn't rename anything. There wasn't the `Screen` type. This doesn't even exist in the old versions, for example 0.2.5 https://docs.rs/crossterm_screen/0.2.5/crossterm_screen/.

File: src/lib.rs (hunk 1 above)
Comment (zrzka) [reply src/lib.rs]: Fixed.

File: src/lib.rs (hunk 2 above)
Comment (zrzka) [reply src/lib.rs]: Fixed.

**Comment by zrzka** [comment]
The `Screen` type was available in the `0.1.0` (https://docs.rs/crossterm_screen/0.1.0/crossterm_screen/) and removed in the `0.2.0` (https://docs.rs/crossterm_screen/0.2.0/crossterm_screen/).

**Comment by TimonPost** [comment]
Yea I went for a search, I forgot that I've already eliminated it because it wasn't needed anymore. Sorry for the confusion. My dependencies in my brain were not updated yet.

**Review by TimonPost — APPROVED** [review]

**Pull request merged**